Lesson2/src/Homework2.java

```diff
@@ -0,0 +1,114 @@
+import java.util.Scanner;
+
+import static java.lang.Math.sqrt;
+
+public class Homework2 {
+
+    public static void main(String[] args) {
+
+        System.out.println(sum(10, 20));
+        System.out.println(sum(Integer.MAX_VALUE, Integer.MAX_VALUE));
+        System.out.println(max(50, 49));
+        System.out.println(average(new int[]{1, 2, 3, 4, 5}));
+        System.out.println(max(new int[]{1, 2, 3, 4, 5, 100, 99}));
+        System.out.println(calculateHypotenuse());
+    }
+
+    /**
+     * 1. Method should return sum of "a" and "b"
+     * 2. (optional) if sum of a and b is more then int can store -- return -1;
+     **/
+    public static int sum(int a, int b) {
+        int x = a + b;
+        int z = 0;
+        long y = Integer.MAX_VALUE + Integer.MAX_VALUE;
```

Comment:
y will be equal to -2, because  Integer.MAX_VALUE + Integer.MAX_VALUE; is sum of two ints, and will be cast to int

if you will run  System.out.println(sum(-10, -20)); from main, you will see that result will be -1 instead of -30

Reply:
to cast the result of summarizing two ints to long you need to cast except one of them to long